rs/nns/sns-wasm/src/sns_wasm.rs

```diff
@@ -1723,6 +1743,56 @@ where
                 )
             })
     }
+
+    // TODO[NNS1-2997]: Remove this function once the migration is completed.
+    pub async fn populate_wasm_metadata(
+        thread_safe_sns: &'static LocalKey<RefCell<SnsWasmCanister<M>>>,
+    ) {
+        // 2. Compute metadata for all marked WASMs.
+        let all_wasm_hashes = thread_safe_sns.with(|sns_canister| {
+            sns_canister
```

Comment:
Since this is going to run on every heartbeat, we probably don't want to create the timers / futures UNLESS there is work to do.  I'd probably do the icheck on the empty metadata here and then produce the list of work to do.  Since it's in a heartbeat, you could just get the next one, if any, that has empty metadata?

Reply:
Also, we might add a thread_local static variable to just tell us if there's anything left to do so we can stop doing any work here.  This all seems a little bit expensive perhaps, given there are under 100 wasms, but we'll be checking them all

Reply:
@jasonz-dfinity  pointed out that the whole work should fit into one message, so I'm going to try to make a sync migration and test that. If that works, we won't need any additional state for the migration.

Reply:
It worked, I tested locally using `bazel test --test_env=SSH_AUTH_SOCK //rs/nns/integration_tests:upgrade_canisters_with_golden_nns_state --test_env=NNS_CANISTER_UPGRADE_SEQUENCE=sns-wasm`